Geometry/HcalAlgo/plugins/DDHCalBarrelAlgo.cc

```diff
@@ -351,84 +318,103 @@ void DDHCalBarrelAlgo::constructGeneralVolume(DDCompactView& cpv) {
+#endif
 
   solid =   DDSolidFactory::polyhedra(DDName(name, idNameSpace),
 				      getNsectortot(), -alpha, dphi, pgonZHalf,
 				      pgonRminHalf, pgonRmaxHalf);
   DDLogicalPart genlogich(DDName(name, idNameSpace), matter, solid);
 
   cpv.position(genlogich, genlogic, 1, r0, rot);
-  LogDebug("HCalGeom") << "DDHCalBarrelAlgo test: "  << genlogich.name() 
-		       << " number 1 positioned in " << genlogic.name() 
-		       << " at " << r0 << " with " << rot;
-
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalBarrelAlgo: "  << genlogich.name() 
+			       << " number 1 positioned in " << genlogic.name()
+			       << " at " << r0 << " with " << rot;
+#endif
   if (getNhalf() != 1) {
     rot = DDRotation(DDName(rotHalf, rotns));
-   cpv.position(genlogich, genlogic, 2, r0, rot);
-    LogDebug("HCalGeom") << "DDHCalBarrelAlgo test:  " << genlogich.name()
-			 << " number 2 positioned in " << genlogic.name()
-			 << " at " << r0 << " with " << rot;
+    cpv.position(genlogich, genlogic, 2, r0, rot);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HCalGeom") << "DDHCalBarrelAlgo: " << genlogich.name()
+				 << " number 2 positioned in " 
+				 << genlogic.name() << " at " << r0 
+				 << " with " << rot;
+#endif
   } //end if (getNhalf...
   
   //Construct sector (from -alpha to +alpha)
   name = idName + "Module";
-  LogDebug("HCalGeom") << "DDHCalBarrelAlgo test: " << DDName(name,idNameSpace)
-		       << " Polyhedra made of " << getGenMaterial() 
-		       << " with 1 sector from " << -alpha/CLHEP::deg << " to "
-		       << alpha/CLHEP::deg << " and with " << nf <<" sections";
-  for (i = 0; i < pgonZHalf.size(); i++) {
-    LogDebug("HCalGeom") << "\t" << "\tZ = " << pgonZHalf[i] << "\tRmin = " 
-			 << pgonRminHalf[i] << "\tRmax = " << pgonRmaxHalf[i];
-  }
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HCalGeom") << "DDHCalBarrelAlgo: "
+			       << DDName(name,idNameSpace)
+			       << " Polyhedra made of " << getGenMaterial() 
+			       << " with 1 sector from "
+			       << convertRadToDeg(-alpha) << " to "
+			       << convertRadToDeg(alpha) << " and with " 
+			       << nf <<" sections";
+  for (unsigned int i = 0; i < pgonZHalf.size(); i++)
+    edm::LogVerbatim("HCalGeom") << "\t" << "\tZ = " << pgonZHalf[i]
+				 << "\tRmin = " << pgonRminHalf[i]
+				 << "\tRmax = " << pgonRmaxHalf[i];
+#endif
 
   solid =   DDSolidFactory::polyhedra(DDName(name, idNameSpace),
 				      1, -alpha, 2*alpha, pgonZHalf,
 				      pgonRminHalf, pgonRmaxHalf);
   DDLogicalPart seclogic(DDName(name, idNameSpace), matter, solid);
   
+  double theta = 90._deg;
   for (int ii=0; ii<getNsectortot(); ii++) {
     double phi    = ii*2*alpha;
-    double phideg = phi/CLHEP::deg;
+    double phideg = convertRadToDeg(phi);
+    double phiy   = phi + 90._deg;
     
     DDRotation rotation;
-    string rotstr("NULL");
-    if (phideg != 0) {
+    std::string rotstr("NULL");
+    if (std::lround(phideg) != 0) {
```

Comment:
One of the motivations for removing CLHEP is to eliminate unnecessary conversions from radians to degrees. Please see how this sort of code is handled in  [DDMuonAngular.cc](https://github.com/cms-sw/cmssw/blob/6383740652b50371bd88dcd809fe8ab28e9c3510/Geometry/MuonCommonData/plugins/DDMuonAngular.cc#L70).
I don't think there is any need to convert angles to degrees except for debug messages.